CMakeLists.txt

```diff
@@ -467,32 +471,31 @@ if(ACTS_BUILD_PLUGIN_GEANT4)
     find_package(Geant4 ${_acts_geant4_version} REQUIRED CONFIG COMPONENTS gdml)
 endif()
 
-if(ACTS_BUILD_PLUGIN_TRACCC)
-    if(ACTS_USE_SYSTEM_ALGEBRAPLUGINS)
-        find_package(algebra-plugins ${_acts_algebraplugins_version} REQUIRED)
-    else()
-        add_subdirectory(thirdparty/algebra-plugins)
-    endif()
-
-    if(ACTS_USE_SYSTEM_DETRAY)
-        find_package(detray ${_acts_detray_version} REQUIRED CONFIG)
+# Convenience macro to simplify the check if we want to setup, and if
+# so, if we want to find it in the system or build internally
+macro(conditional_setup _var _package _path)
+    if(ACTS_SETUP_${_var})
+        if(ACTS_USE_SYSTEM_${_var})
+            message(VERBOSE "Use system ${_package}")
+            find_package(${_package} ${ARGN})
+        else()
+            message(VERBOSE "Use internal ${_package}")
+            add_subdirectory(${_path})
+        endif()
     else()
-        add_subdirectory(thirdparty/detray)
+        message(VERBOSE "Don't setup ${_package} explicitly")
     endif()
+endmacro()
 
-    if(ACTS_USE_SYSTEM_VECMEM)
-        find_package(vecmem ${_acts_vecmem_version} REQUIRED)
-    else()
-        add_subdirectory(thirdparty/vecmem)
-        # Make the "VecMem language code" available for the whole project.
+if(ACTS_BUILD_PLUGIN_TRACCC)
+    conditional_setup(ALGEBRAPLUGINS algebra-plugins thirdparty/algebra-plugins _acts_algebraplugins_version REQUIRED)
+    conditional_setup(DETRAY detray thirdparty/detray _acts_detray_version REQUIRED CONFIG)
+    conditional_setup(VECMEM vecmem thirdparty/vecmem _acts_vecmem_version REQUIRED)
```

Comment:
Don't hide this behind `ACTS_BUILD_PLUGIN_TRACCC`!  Move the setup of the externals back to the top level.

In general the setup of externals, and the decision about which plugins are built, are independent of each other. By combining the logic like this, you'll just invite trouble down the road.

Granted, the old SYCL plugin (which used to use vecmem) has been removed some time ago. But we could absolutely have more elaborate relationships between externals and plugins than the one proposed here.)

Reply:
Yeah so I just took this from the previous state. So the plugins where set up only within `ACTS_BUILD_PLUGIN_TRACCC` before...
So currently, the whole cmake is set up to build either everything from the traccc-universe or nothing, like in `Plugins/CMakeLists.txt`:
```
add_component_if(Detray PluginDetray ACTS_BUILD_PLUGIN_TRACCC)
add_component_if(Covfie PluginCovfie ACTS_BUILD_PLUGIN_TRACCC)
```
So they are all triggered by `ACTS_BUILD_PLUGIN_TRACCC`.

I think I would not change/generalize this in that PR, since it seems this granularity is not needed right now. Maybe we can disentangle this later?

Reply:
You're right. I didn't check, the current logic is indeed like this. :thinking: I believe I may have complained about this also when it was first set up.

Okay, let's keep the current logic then. But I still think that the logic in this file should be reduced in the future, and valid configurations could be provided to the users through presets. Unfortunately there's no golden path here, as this many flags will never be possible to set up such that the user couldn't set them to an invalid combination. We should (I believe) just make sure that technically as many combinations can be expressed using the flags as possible.

Reply:
Just chiming in here that I think we should stick to the all-or-nothing configuration to reduce complexity. 

In my opinion, both

- whether we use traccc et al at all
- if we use fetch content or find package 

Should be behind a single flag each. We should not introduce additional granularity and I advocated for not having individual plugins and flags for the sub projects.